hazelcast/src/main/java/com/hazelcast/map/impl/MetadataInitializer.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.nio.serialization.Data;
+
+import java.io.IOException;
+
+public interface MetadataInitializer {
```

Comment:
cant we use `RecordStoreMutationObserver` instead?

Reply:
Looks like a good idea. However RecordStoreMutationObserver passes object format of new values to update method. We need the binary format when the in memory format is binary.

Reply:
`record.getValue()` should help here if raw storage format is needed.

Reply:
`record.getValue()` returns the old value during update operations.

Reply:
if you see it is possible to adapt `RecordStoreMutationObserver` to your needs, it can be better to not to pollute code with a similar interface. Maybe it can be done in another PR.